src/graph/network.cc

```diff
@@ -8,19 +8,29 @@
 #include "./network/communicator.h"
 #include "./network/socket_communicator.h"
 #include "./network/serialize.h"
+#include "./network/common.h"
 
 #include "../c_api_common.h"
 
+#include <unordered_map>
+
 using dgl::runtime::DGLArgs;
 using dgl::runtime::DGLArgValue;
 using dgl::runtime::DGLRetValue;
 using dgl::runtime::PackedFunc;
 using dgl::runtime::NDArray;
+using dgl::network::StringPrintf;
 
 namespace dgl {
 namespace network {
 
-// Wrapper for Send api
+static std::unordered_map<void*, char*> MemoryPool;
```

Comment:
Check out memory pool implementation in [dmlc-core/memory.h](https://github.com/dmlc/dmlc-core/blob/master/include/dmlc/memory.h). Can we reuse that?

Reply:
can we have a thread-safe memory pool?

Reply:
Actually, we don't need a memory pool here. We just need to a container that stores the pointer of the data  buffer.